client/asset/dcr/dcr.go

```diff
@@ -1226,47 +1240,21 @@ func (dcr *ExchangeWallet) spendableUTXOs(unspents []walletjson.ListUnspentResul
 			utxos = append(utxos, &compositeUTXO{
 				rpc:   txout,
 				input: nfo,
+				confs: txout.Confirmations,
 			})
 			sum += toAtoms(txout.Amount)
+			if confs == 0 {
+				unconf += toAtoms(txout.Amount)
+			}
 		} else {
 			unconf += toAtoms(txout.Amount)
 		}
 	}
 	return utxos, sum, unconf, nil
 }
 
-// balances creates a slice of *asset.Balance corresponding to the input slice
-// of minconf.
-func (dcr *ExchangeWallet) balances(confs []uint32) ([]*asset.Balance, error) {
-	if len(confs) == 0 {
-		return nil, fmt.Errorf("balances confs slice cannot be empty")
-	}
-	unspents, err := dcr.node.ListUnspentMin(0)
-	if err != nil {
-		return nil, err
-	}
-	balances := make([]*asset.Balance, len(confs))
-	for i := range balances {
-		balances[i] = new(asset.Balance)
-	}
-	for _, txout := range unspents {
-		txHash, err := chainhash.NewHashFromStr(txout.TxID)
-		if err != nil {
-			return nil, fmt.Errorf("error decoding txid from rpc server %s: %v", txout.TxID, err)
-		}
-		for i, balance := range balances {
-			// Change from a DEX-monitored transaction is exempt from the fundconf requirement.
-			if txout.Confirmations >= int64(confs[i]) || dcr.isDEXChange(txHash, txout.Vout) {
-				balance.Available += toAtoms(txout.Amount)
-			} else {
-				balance.Immature += toAtoms(txout.Amount)
-			}
-		}
-	}
-	return balances, nil
-}
-
 // lockedAtoms is the total value of locked outputs, as locked with LockUnspent.
+// TODO: handle multiple accounts!
```

Comment:
It occurred to me that Balance.Locked amount is going to include locked coins from all accounts, which is misleading.